codemods/run-codemod.ts

```diff
@@ -0,0 +1,191 @@
+        return fs.lstatSync(path.join(codemodsDir, folder)).isDirectory();
+    });
+
+    const options = codemods.map((mod) => ({
+        name: mod,
+        value: mod,
+        description: codemodDescriptions[mod as Codemod],
+    }));
+
+    return options;
+}
+
+function runCodemods(
+    codemodsToRun: string[],
+    outputPath: string,
+    options?: { "migrate-colour"?: Theme }
+): void {
+    codemodsToRun.forEach((codemod) => {
+        const codemodPath = path.join(codemodsDir, codemod, "index.ts");
+        let command = `jscodeshift --parser=tsx -t ${codemodPath} ${outputPath}`;
+
+        if (codemod === Codemod.MigrateColour && options?.["migrate-colour"]) {
+            command = `jscodeshift --parser=tsx -t ${codemodPath} --mapping=${options["migrate-colour"]} ${outputPath}`;
+        }
+
+        console.log(
+            `Running codemod: ${codemod} on output path: ${outputPath}`
+        );
+
+        try {
+            execSync(command, { stdio: "inherit" });
+            console.log(
+                `Codemod ${codemod} executed successfully on ${outputPath}.`
+            );
+        } catch (error) {
+            console.error(
+                `Error executing codemod ${codemod}: ${error.message}`
+            );
+        }
+    });
+}
+
+// Choose path for codemod use
+async function chooseOutputPath(): Promise<string | null> {
+    const options = [
+        { name: "src (./src)", value: predefinedPaths.Src },
+        { name: "current directory (./)", value: predefinedPaths.Codebase },
+        { name: "custom", value: "custom" },
+    ];
+
+    const selectedOption = await select({
+        message: "Choose an output path:",
+        choices: options,
+    });
+
+    if (selectedOption === "custom") {
+        const customPath = await input({
+            message: "Enter your custom path:",
+        });
+
+        if (!customPath) {
+            console.log("No custom path provided. Canceling.");
+            return null;
+        }
+
+        const resolvedPath = path.resolve(customPath);
+
+        if (!fs.existsSync(resolvedPath)) {
+            console.error(
+                `The path "${resolvedPath}" does not exist. Please try again.`
+            );
+            return null;
+        }
+
+        return resolvedPath;
+    }
+
+    return selectedOption;
+}
+
+async function chooseTheme(): Promise<Theme | null> {
+    const themeOptions = [
+        { name: "LifeSG", value: Theme.LifeSG },
+        { name: "BookingSG", value: Theme.BookingSG },
+    ];
+
+    const selectedTheme = await select({
+        message: "Select the theme to follow for Color migration:",
+        choices: themeOptions,
+    });
+
+    return selectedTheme;
+}
+
+// Choose and run the codemods
+async function main(): Promise<void> {
+    const codemods = listCodemods();
+    if (codemods.length === 0) {
+        console.log("No codemod scripts found.");
+        return;
+    }
+
+    try {
```

Comment:
to avoid deeply nested code, what i normally do is to put the main logic in a function, and then wrap it in a try catch

```
function getOptions() {
 // ...
}

function main() {
  try {
    const options = getOptions()
    runCodemods(...)
  } catch (error) {
     // error handling
  }
}
```